### ISSUE TITLE
Ideals in generic polynomial rings

### ISSUE THREAD

**Issue by HechtiDerLachs** [oscar-system/Oscar.jl#988]
```julia
R, (x,y) = QQ["x", "y"]
Y = Spec(R)
Q = OO(Y)
T, _ = PolynomialRing(Q, ["u", "v"])

I = ideal(T, [Q(x)])
```
doesn't work, i.e. I *can not define* an ideal in the ring `T`. I understand that things like ideal membership, etc. needs special care. But I would really like to be able to define any finitely generated ideal in any polynomial ring. The same applies for graded rings, i.e. if I was using 
```julia
S, (u,v) = grade(T, [1,1])
```
instead of `T`.

Edit: The point is that the error message complains about a missing implementation of `singular_ring` for arguments of type `MPolyQuoLocalizedRing`. This can not be solved directly, since there simply does not exist anything on the singular side for these objects. In my opinion, the mistake comes from trying to store the generators of any `MPolyIdeal` in a `BiPolyArray`. This code is simply not generic and therefore `MPolyIdeal` fails to serve its purpose here.

**Comment by fieker** [comment]
On Mon, Jan 17, 2022 at 10:53:20AM -0800, Matthias Zach wrote:
> ```julia
> R, (x,y) = QQ["x", "y"]
> Y = Spec(R)
> Q = OO(Y)
> T, _ = PolynomialRing(Q, ["u", "v"])
> 
> I = ideal(T, [Q(x)])
> ```
> doesn't work, i.e. I *can not define* an ideal in the ring `T`. I understand that things like ideal membership, etc. needs special care. But I would really like to be able to define any finitely generated ideal in any polynomial ring. The same applies for graded rings, i.e. if I was using 
> ```julia
> S, (u,v) = grade(T, [1,1])
> ```
> instead of `T`.
> 
The whole module is under development and incomplete.
There is no chance whatsoever to deal with all polynomial rings. I think
intended here are ideals over multivariate rings.

When I try to run this snipped I get
...
ERROR: MethodError: no method matching
groebner_basis(::Oscar.BiPolyArray{gfp_mpoly}; ord=:degrevlex,
complete_reduction=true)
Closest candidates are:
  groebner_basis(::Oscar.BiPolyArray, ::MonomialOrdering;
  complete_reduction) at ~/.julia/dev/Oscar/src/Rings/groebner.jl:455
  got unsupported keyword argument "ord"
  groebner_basis(::Oscar.BiPolyArray; ordering, complete_reduction) at
  ~/.julia/dev/Oscar/src/Rings/groebner.jl:69 got unsupported keyword
  argument "ord"
  groebner_basis(::Oscar.ModuleGens) at
  ~/.julia/dev/Oscar/src/Modules/UngradedModules.jl:2150 got
  unsupported keyword arguments "ord", "complete_reduction"
  ...

Which is part of a bigger problem. The "entire" groebner_basis/
groebner_assure stack was partly rewritten and thus is now incomplete
(and partly broken).

I find it actually disturbing that printing (or creation of an ideal) is
already triggereing groebner calls.

So: go Geometry Task Force!

**Comment by HechtiDerLachs** [comment]
Are you sure you run the snippet I mentioned above? There are no `gfp_mpoly`s involved.

**Comment by fieker** [comment]
On Tue, Jan 18, 2022 at 12:44:29AM -0800, Matthias Zach wrote:
> Are you sure you run the snippet I mentioned above? There are no `gfp_mpoly`s involved. 
> 
Maybe its a different Oscar version, but yes, I did run it as posted...

**Comment by HechtiDerLachs** [comment]
Just to push this issue a little; this problem has a long list of repercussions. For instance, I also can not define a homomorphism of rings such as `S` or `T` above. So for now, I need to introduce a long list of workarounds that will need to be removed later on. Who is responsible for this part of code? Who can I talk to about this issue?

@wdecker , @jankoboehm , @afkafkafk13 : Any suggestions?

**Comment by thofma** [comment]
@ederc We talked about delaying any Singular invocation to when it really is necessary, right? I think this would help here too. At least then one can construct the ideals.

**Comment by thofma** [comment]
This is fixed by #998.

**Comment by ederc** [comment]
Yes, #998 will fix it. Working on it.

**Comment by HechtiDerLachs** [comment]
That is still work in progress, then? Let me know when it's ready!

**Comment by thofma** [comment]
Fixed by #998.
